packages/shared/src/hooks/vote/types.ts

```diff
@@ -1,81 +1,139 @@
-import { Post, ReadHistoryPost, UserPostVote } from '../../graphql/posts';
+import { MutationKey } from '@tanstack/react-query';
+import {
+  Post as PostEntity,
+  ReadHistoryPost,
+  UserVote,
+} from '../../graphql/posts';
 import { Origin } from '../../lib/analytics';
 import { PostAnalyticsEventFnOptions } from '../../lib/feed';
 import { UseMutationMatcher } from '../mutationSubscription/types';
 
-export type ToggleVoteProps = {
+export type VoteEntityPayload = {
+  id: string;
+  userState?: {
+    vote: UserVote;
+  };
+};
+
+export type ToggleVoteProps<T extends VoteEntityPayload = VoteEntityPayload> = {
   origin: Origin;
-  post: Post | ReadHistoryPost;
+  payload: T;
+  entity: UserVoteEntity;
   opts?: PostAnalyticsEventFnOptions;
 };
 
 export type VoteProps = {
   id: string;
+  entity: UserVoteEntity;
+};
+
+export type UseVote = {
+  upvote: (props: VoteProps) => Promise<void>;
+  downvote: (props: VoteProps) => Promise<void>;
+  cancelVote: (props: VoteProps) => Promise<void>;
+  toggleUpvote: (
+    props: ToggleVoteProps<PostEntity | ReadHistoryPost>,
```

Comment:
Do we need the explicit entities here? (is the idea to add more as we add them?)

Reply:
Currently it is like this since we split post types. This one is specific to `useVotePost`. So at this point just legacy but can't clean it up fast. `ToggleVoteProps` can accept any future entity that supports voting.